content/stable-typescript-types-in-ember-5-1.md

```diff
@@ -0,0 +1,359 @@
+
+    - While many Ember users *do* rely on Ember Data, not all do, and this makes the best possible types for Ember available to anyone who is not blocked in this way.
+
+    - Ember Data itself can now take advantage of the Ember types! This was a long-standing chicken-or-egg problem: Ember Data could not really publish its own stable types while relying on the hand-written `@types` packages for Ember!
+
+    Happily, the two newest members of the Ember TypeScript team, Peter Wagenet and Krystan HuffMenne, are both actively interested in tackling this problem, so there will hopefully be good progress in this space in the rest of 2023. (This is a great place to chip in and help, by the way!)
+
+2. Ember's stable types do not have full support for the Ember Classic programming model. If you rely heavily on the Ember Classic programming model, we encourage you to update your code to the Ember Octane programming model—for *many* reasons! For details on absorbing these changes in particular, see the [Migration Guide](#toc_migration-guide) below.
+
+
+## Stability
+
+Because of the interop challenge with Ember Data, our TypeScript support is *Released* but not yet *Recommended*. This is one of the first major features to land in Ember which leans on this distinction in a way that matters to end users, so here is a quick explainer. Since [RFC 0617: RFC Stages][rfc-0617], features go through up to six stages on the way to being “done”:
+
+0. **Proposed:** A proposal for a change to Ember or its processes that is offered for community and team evaluation.
+1. **Exploring:** An RFC deemed worth pursuing but in need of refinement.
+2. **Accepted:** A fully specified RFC. Waiting for or in the process of implementation.
+3. **Ready for Release:** The implementation of the RFC is complete, including learning materials.
+4. **Released:** The work is published. If it is codebase-related work, it is in a stable version of the relevant package(s).
+5. **Recommended:** The feature/resource is recommended for general use.
+
+Since TypeScript is *Released*, you can rely on all of our normal SemVer guarantees. Ember's types are now part of our public contract with you as a user, and they have to be maintained with the exact same care as any other part of the framework. Additionally, Chris Krycho and the rest of the Ember TypeScript team developed [a spec for Semantic Versioning for TypeScript Types][semver-ts], which allows Ember (and any other framework or library!) to define how it manages supported TypeScript versions and absorb breaking changes from TypeScript. That means it is totally safe for you to rely on these types if you are not impacted by the caveats described above!
+
```

Comment:
```suggestion

One specific point: as described on our [Releases](https://emberjs.com/releases/) page, the Ember framework itself uses a “rolling window” support policy for TypeScript versions. This means Ember can bump its minimum supported TypeScript version in an Ember minor release, just like it can for Node.js versions—but the supported range of TypeScript versions across two Ember LTS releases must always overlap. So if Ember 5.4 LTS supports TypeScript 4.9–5.3, Ember 5.8 LTS *could* drop support for TypeScript versions earlier than TypeScript 5.3, but must include at least TypeScript 5.3. That guarantees that users can always upgrade their TypeScript and Ember versions separately. In practice, we expect to maintain a fairly wide TypeScript version support window to minimize the impact on the ecosystem—likely (though not guaranteed to be) 4 to 6 recent TypeScript versions at any given time.

Ember's initial supported TypeScript version range is TypeScript 4.9, 5.0, and 5.1.

```